hil/src/ssh_wrapper.rs

```diff
@@ -0,0 +1,127 @@
+use color_eyre::{eyre::bail, Result};
+use ssh2::Session as Ssh2Session;
+use std::io::Read;
+use std::net::TcpStream;
+use tracing::{debug, info};
+
+/// SSH wrapper that supports both password and key authentication
+/// Will be removed in favor of ssh keys
+pub struct SshWrapper {
+    session: Ssh2Session,
+}
+
+impl SshWrapper {
+    pub async fn connect(
+        host: String,
+        port: u16,
+        username: String,
+        password: String,
+    ) -> Result<Self> {
+        info!("Connecting to {}@{}:{}", username, host, port);
+
+        let tcp = TcpStream::connect(format!("{host}:{port}")).map_err(|e| {
+            color_eyre::eyre::eyre!("Failed to connect to SSH server: {}", e)
+        })?;
+
+        let mut session = Ssh2Session::new().map_err(|e| {
+            color_eyre::eyre::eyre!("Failed to create SSH session: {}", e)
+        })?;
+
+        session.set_tcp_stream(tcp);
+        session.handshake().map_err(|e| {
+            color_eyre::eyre::eyre!("Failed to perform SSH handshake: {}", e)
+        })?;
+
+        session
+            .userauth_password(&username, &password)
+            .map_err(|e| {
+                color_eyre::eyre::eyre!("SSH password authentication failed: {}", e)
+            })?;
+
+        if !session.authenticated() {
+            bail!("SSH authentication failed");
+        }
```

Comment:
you're doing a ton of blocking work in async. Never ever block in async, use tokio::task::spawn_blocking